sv.c

```diff
@@ -1496,6 +1507,42 @@ Perl_sv_upgrade(pTHX_ SV *const sv, svtype new_type)
     }
 }
 
+struct xpvhv_aux*
+Perl_hv_auxalloc(pTHX_ HV *hv) {
+    const struct body_details *old_type_details = bodies_by_type + SVt_PVHV;
+    void *old_body;
+    void *new_body;
+
+    PERL_ARGS_ASSERT_HV_AUXALLOC;
+    assert(SvTYPE(hv) == SVt_PVHV);
+    assert(!SvOOK(hv));
+
+#ifdef PURIFY
+    new_body = new_NOARENAZ(&fake_hv_with_aux);
+#else
+    new_body_inline(new_body, SVt_IV, fake_hv_with_aux);
```

Comment:
I think this could do with a comment to explain the `SVt_IV`. I suspect it somehow links with the renaming of that parameter from `svtype` to `root_index` (in the macro, but not the function), but I don't really have a clue. (And I'm not sure `new_body_inline` is still the most appropriate name for the macro.)

Reply:
> I think this could do with a comment to explain the `SVt_IV`. I suspect it somehow links with the renaming of that parameter from `svtype` to `root_index` (in the macro, but not the function), but I don't really have a clue. (And I'm not sure `new_body_inline` is still the most appropriate name for the macro.)

`SVt_IV` became `#  define HVAUX_ARENA_ROOT_IX   SVt_IV` in `sv.h` and more comments.

Yes, `new_body_inline` wasn't really the best name. It's now `new_body_from_arena` - if there's a better name still, it eludes me.

I also found some outdated comments in `sv.c` to eliminate.